eng/targets/ResolveReferences.targets

```diff
@@ -144,7 +144,7 @@
     This executes on NuGet restore and during DesignTimeBuild. It should not run in the outer, cross-targeting build.
    -->
   <Target Name="ResolveCustomReferences"
-      BeforeTargets="CollectPackageReferences;ResolveAssemblyReferencesDesignTime;ResolveAssemblyReferences"
+      BeforeTargets="CheckForImplicitPackageReferenceOverrides;CollectPackageReferences;ResolvePackageAssets"
```

Comment:
/fyi there were two things wrong w/ the old `BeforeTargets` value:
1. `ResolveAssemblyReferencesDesignTime` didn't matter because that depends on `ResolveAssemblyReferences`
2. `ResolveAssemblyReferences` runs later (closer to `CoreCompile`) than needed to ensure information needed from packages is available where / when it needs to be